src/chain/store/index.rs

```diff
@@ -145,6 +146,46 @@ impl<DB: Blockstore> ChainIndex<DB> {
             })
         })
     }
+
+    /// Finds the latest beacon entry given a tipset up to 20 tipsets behind
+    pub fn latest_beacon_entry(&self, ts: &Tipset) -> Result<BeaconEntry, Error> {
```

Comment:
maybe we can use `anyhow` here, this `Error::Other` is not really more helpful. We might want to do it in a separate PR, though.

Reply:
Error handling will be cleaned up separately.